### DISCUSSION TITLE
Confusion with `spawn` and service scopes

### DISCUSSION THREAD

**Discussion by UberMouse** [statelyai/xstate#1426] [General]
I'm trying to prototype a new UI architecture for XState + React I want to try at work. Basically attaching views to machines and providing invoked/spawned machines as "slots" to the machines view. I've got a prototype working for invoked machines but I'm struggling to get it working for `spawn`'ed machines.

As you can see here, https://codesandbox.io/s/heuristic-payne-1p3t5?file=/src/index.js:3649-3658, I'm spawning an actor inside my top level machine. The problem I'm getting is that it's ending up as a deferred actor instead of an interpreter which means I can't access the meta value of the machine reliably. You can see the actor logged out to the console with `deferred: true` set

Really mystified as to why. I'm spawning it inside of an assignment call so it should correctly be in the scope of its parent machine? It's also a source of complexity that invoked machines end up in `state.children` and the spawned actor does not, from looking at the code of `interpreter.spawn` it should store spawned machines in `state.children`, it's just not because the actor is deferred?

**Comment by davidkpiano** [comment]
> I can't access the meta value of the machine reliably

What do you mean by the meta value of the machine? Can you give specific steps to reproduce your problem?

**Reply by UberMouse** [reply]
That's unrelated to this problem. I just mean that accessing meta indirectly via the deferred actor is a pain because the meta object I can reach (I can't remember through which path from a deferred actor) has the meta keyed by the id of the machine so it's 
`meta: { machinId: { ...machineMeta } }` whereas accessing it via the interpreter is `meta: { ...machineMeta }`. One of those paths requires me to know the machine id and one doesn't. (sorry I don't remember all the details)

I saw that invoking a machine in the initial state of the first machine passed to the interpreter resulted in it being deferred as well, is that the expected behaviour? I worked around it by setting the initial state to be a state that always transitions to the "real" initial state and the invoked machine was no longer deferred. I suspect that's the same problem I encountered here, not something specific to `spawn`, as I was trying to `spawn` the actor in the initial state of the machine.
